Create CCC-TAC-Minutes-2023-05-18.md (#172)
confidential-computing/governance: TAC/Meetings/2023/2023-05-18/CCC-TAC-Minutes-2023-05-18.md

* Ravi Sahita (Rivos)
* Drasko Draskovic (Ultraviolet)
* Xinxin Fan (IoTeX)
* Sonemaly Phrasavath (AMD)
* Mona Vij (Intel)
* Shanwei Cen (Intel)
* Kurt Taylor (LF)
* Helen Lau (LF)

**Outreachy internship**

* Zahara Tarkhani introduced herself and discussed the hardware needed for the mentorship
* All agreed to continue the discussion via email

**Meeting Minute Approval**

Proposed: That the minutes of the May 4, 2023 meeting of the Technical Advisory Council of the Confidential Computing Consortium as distributed to the members of the TAC in advance of this meeting are hereby adopted and approved.

* Quorum was reached and the minutes were approved by the voting members of the TAC. 

**Board Meeting Recap**

* DM briefly recapped the Governing Board meeting also held this week

**Project progression discussion**

* DM introduced the discussion on fixing the current project progression policy 
 * Project Progression proposal [PR #160](https://github.com/confidential-computing/governance/pull/160)
* Dave Thaler (DT) presented on the Graduation stage
  * Background: [Issue #104](https://github.com/confidential-computing/governance/issues/104) from April 2022

>   * Sandbox and Incubation stages were approved soon after initial project formation, but further stages were never defined or approved

Reviewer: ```suggestion
  * Sandbox and Incubation stages were approved soon after initial project formation, but further stages were never discussed or approved
```

  * Need is to define the differences and benefits of Graduation
  * One approach is to remove it alltogether [PR #171](https://github.com/confidential-computing/governance/pull/171)

>   * The other approach is to define the Graduation phase [PR #170](https://github.com/confidential-computing/governance/pull/170)

Reviewer: ```suggestion
  * The other approach is to update the Graduation phase text to address points raised [PR #170](https://github.com/confidential-computing/governance/pull/170)
```

>     * Emeritus not addressed in this proposal

Reviewer: ```suggestion
    * Emeritus not addressed in this discussion
```

    * [Growth Plan document](https://github.com/confidential-computing/governance/blob/main/growth-plans.md)

>     * Discussion on employer wording - proposed that employer would include subsidaries and contractors, what fraction affiliated with project

Reviewer: ```suggestion
    * Discussion on employer wording - proposed that employer would include subsidaries and contractors for consistency with CCC voting membership policies, what fraction affiliated with project
```

>     * Numer of committers - make more clear than only the subjective term "healthy number"

Reviewer: ```suggestion
    * Number of committers - make more clear than only the subjective term "healthy number"
```

    * OpenSSF Best Practices   https://bestpractices.coreinfrastructure.org/en/criteria
        * From OpenSSF "Bus factor "- minimum number of maintainers that would have to suddenly disappear for the project to stall
        * In chat: Independent of this, I would love to see the TAC recommend OpenSSF Best Practices badge for all projects.  Can discuss next time.
  * Mike Bursell proposed that the TAC debate the 2 options, what the levels are for: example - would the funding be higher or lower for graduation
* DM asked the TAC to review the proposals for discussion at the next meeting

**Tech Talk: Keylime**

* Thore Sommer (TS) presented on the Keylime (CNCF project)
  * Background on TPM - Trusted Platform Module
    * PCR extend algotithm
  * Questions and discussion on measurement
  * Keylime architecture and example of process at runtime
  * SEV-SNP vTPM proof-of-concept: https://github.com/svsm-vtpm/
  * Plans:
      * move UEFI log parser to pure Python, evaluating policy engines
      * General attestation infrastructure
  * Work with CCC for common terminology and possible hosting plugin, possible additional future modules

>     * DT discussed the plugin hosted in the CCC would be similar to Veraison

Reviewer: ```suggestion
    * DT asked if TPM based attestation starts with an immutable source of measurements or only immutable firmware, since if the latter, then no guarantee that all measurements aren't lies if firmware is compromised.  TS will follow up to verify.
    * DT discussed the plugin hosted in the CCC would be similar to what Veraison would need, since Keylime is in the same overall space as Veraison
```

    * MB asked if attestation - is there a danger in combining TEE and TPM attestation, pointing out the differences between them - discussion ensued
    * DM asked about other areas of use or examples
    * From chat: For the CRTM (initial measurment) (definition)]https://trustedcomputinggroup.org/wp-content/uploads/TCG_PCClient_PFP_r1p05_v23_pub.pdf)
    * DT asked about the attestation architecture - relying party needing to be able to tell they are a TEE

**TAC Goal Topic: Community Health**

* Nick Vidal (NV) and DM on Community Health
* DM presented on DCI
  * Requirements, Recommendations and Resources (as a part of Minimum Viable Governance)
  * Review of 2022 DCI accomplishments
    * Linux Foundation [Diversity and Inclusivity Courses](https://www.linuxfoundation.org/about/diversity-inclusivity)
* NV discussed the examples from the Outreach committee
  * Outreachy - mentoring: Enarx and Veracruz have had previous mentees
  * New Vice Chair - Kate George
  * Highlighted International Women's Day

**Update: Attestation SIG**

* Shanwei Cen (SC) presented
* No change in the SIG mission or scope

>   * DT asked if the mission would need to changed since they may not actually be developing software

Reviewer: ```suggestion
  * DT asked if the mission statement should be updated since the SIG may not actually be developing software.  Instead it is doing documentation and coordination among other projects (whether in or out of the CCC).
```

* DM welcomed the members of the TAC and reviewed the values of the CCC and the antitrust policy of the Linux Foundation
* Kurt Taylor (KT) assisted with the meeting minutes
* DM reviewed the agenda

### Roll Call / Attendance

***Voting members of the TAC***

*Per the [charter](https://charter.confidentialcomputing.io), all [CCC Premier members](https://confidentialcomputing.io/members/) receive one vote on the TAC. Quorum for votes is at least 50% of voting members present.*

* Accenture - Giuseppe Giordano 
* Arm - Thomas Fossati + / Michael Lu
* Google - Cfir Cohen + / Catalin Sandu
* Huawei - Zhipeng (Howard) Huang
* Intel - Dan Middleton (TAC chair) + / Simon Johnson
* Meta - Eric Northup / Shankaran Gnanashanmugam
* Microsoft - Dave Thaler +
* Red Hat - Lily Sturmann  / Yash Mankad +

   " + " indicates attending

***Other Attendees***

* Mike Bursell (CCC)
* Thore Sommer (Keylime/FHNW)
* Ofir Azoulay-Rozanes (Anjuna)
* Mark Novak (JP Morgan Chase)
* Leonardo Garcia (Linaro)
* Pawan Khandavilli (Microsoft)
* Nick Vidal
* Kevin Zhao (Linaro)
* Marc Meunier (Arm)
* Vidhya Krishnan (NVIDIA)
* Neil Vexler (Anjuna)
* Alec Fernandez (Microsoft)
* Eric Voit (Cisco)

> * Zahara Tarkhani (Microsoft)

Reviewer: ```suggestion
* Zahra Tarkhani (Microsoft)
```

* Ravi Sahita (Rivos)
* Drasko Draskovic (Ultraviolet)
* Xinxin Fan (IoTeX)
* Sonemaly Phrasavath (AMD)
* Mona Vij (Intel)
* Shanwei Cen (Intel)
* Kurt Taylor (LF)
* Helen Lau (LF)

**Outreachy internship**

* Zahara Tarkhani introduced herself and discussed the hardware needed for the mentorship
* All agreed to continue the discussion via email

**Meeting Minute Approval**

Proposed: That the minutes of the May 4, 2023 meeting of the Technical Advisory Council of the Confidential Computing Consortium as distributed to the members of the TAC in advance of this meeting are hereby adopted and approved.

* Quorum was reached and the minutes were approved by the voting members of the TAC. 

**Board Meeting Recap**

* DM briefly recapped the Governing Board meeting also held this week

**Project progression discussion**

* DM introduced the discussion on fixing the current project progression policy 
 * Project Progression proposal [PR #160](https://github.com/confidential-computing/governance/pull/160)
* Dave Thaler (DT) presented on the Graduation stage
  * Background: [Issue #104](https://github.com/confidential-computing/governance/issues/104) from April 2022

>   * Sandbox and Incubation stages were approved soon after initial project formation, but further stages were never discussed or approved

Reviewer: ```suggestion
  * Sandbox and Incubation stages were approved soon after initial CCC formation, but further stages were never discussed or approved
```

    * OpenSSF Best Practices   https://bestpractices.coreinfrastructure.org/en/criteria
        * From OpenSSF "Bus factor "- minimum number of maintainers that would have to suddenly disappear for the project to stall
        * In chat: Independent of this, I would love to see the TAC recommend OpenSSF Best Practices badge for all projects.  Can discuss next time.
  * Mike Bursell proposed that the TAC debate the 2 options, what the levels are for: example - would the funding be higher or lower for graduation
* DM asked the TAC to review the proposals for discussion at the next meeting

**Tech Talk: Keylime**

* Thore Sommer (TS) presented on the Keylime (CNCF project)
  * Background on TPM - Trusted Platform Module
    * PCR extend algotithm
  * Questions and discussion on measurement
  * Keylime architecture and example of process at runtime
  * SEV-SNP vTPM proof-of-concept: https://github.com/svsm-vtpm/
  * Plans:
      * move UEFI log parser to pure Python, evaluating policy engines
      * General attestation infrastructure
  * Work with CCC for common terminology and possible hosting plugin, possible additional future modules

>     * DT asked if TPM based attestation starts with an immutable source of measurements or only immutable firmware, since if the latter, then no guarantee that all measurements aren't lies if firmware is compromised.  TS will follow up to verify.

Reviewer: ```suggestion
    * DT asked if TPM based attestation starts with an immutable source of measurements or only mutable firmware, since if the latter, then no guarantee that all measurements aren't lies if firmware is compromised.  TS will follow up to verify.
```
